### PR TITLE
feat: simpler AccordionPane

### LINKED ISSUE #1224: AccordionPane improvements
**Enhancement**

The `AccordionPane` expects a child renderer that returns `TitlePane` widgets, but these have to manually apply many renderer-supplied props. We should make `AccordionPane` return a `Pane` to each renderer instead of additional props, to make the use of `AccordionPane` easier.

Package Version: `7.0.0`

### LINKED ISSUE #1345: Allow title-pane to be controlled
## Proposal

As per issue #1330 we are adding in a controlled version of each of our `initial*` properties. These will allow the widget to be fully controlled should the user wish. 

## New Properties

- `open?: boolean`

### PR DESCRIPTION
**Pull request by bitpshr**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [x] Unit tests are included in the PR
* [x] For new widgets, an entry has been added to the `.dojorc`
* [x] For new widgets, `theme.variant()` is added to the root domnode
* [x] Any widget variant uses `theme.compose` like [this](https://github.com/dojo/widgets/issues/847)
* [x] WidgetProperties are exported

**Description:**

This pull request adds a new supporting `Pane` widget that's exported from the `Accordion` class, meant to be used in place of raw `TitlePane` children. This mitigates the need to pass a `theme` to the child renderer to align with the latest widget conventions.

Resolves #1224 
Resolves #1345

### CODE REVIEW

File: .dojorc [hunk 1]
```diff
@@ -6,7 +6,7 @@
 	"build-widget": {
 		"prefix": "dojo",
 		"widgets": [
-			"src/accordion-pane",
+			"src/accordion",
```
Comment (bitpshr) [inline .dojorc:9]: `AccordionPane` was renamed `Accordion` throughout, as this felt more appropriate now that a `Pane` is also exported.

File: src/title-pane/index.tsx [hunk 2]
```diff
@@ -15,6 +15,8 @@ export interface TitlePaneProperties {
 	headingLevel?: number;
 	/** If true the pane is opened and content is visible initially */
 	initialOpen?: boolean;
+	/** Explicitly control TitlePane */
+	open?: boolean;
```
Comment (tomdye) [inline src/title-pane/index.tsx:19]: you should be using `initialOpen` here to control it, not `open`.

Comment (bitpshr) [reply src/title-pane/index.tsx:19]: @tomdye per our discussion, this should be ready for a final review as-is. Thanks in advance.

File: src/title-pane/index.tsx [hunk 3]
```diff
@@ -55,19 +57,12 @@ export const TitlePane = factory(function TitlePane({
...
 		onOpen,
+		open: openProp,
 		theme: themeProp
 	} = properties();
 	const { content, title } = children()[0];
 
-	let open = icache.get('open');
-	let performTransition = false;
-	const existingInitialOpen = icache.get('initialOpen');
-	if (existingInitialOpen !== initialOpen) {
```
Comment (tomdye) [inline src/title-pane/index.tsx:65]: I believe you'd still need the `initialOpen` equality check in here would you not?

**Pull request merged**